### PR TITLE
[Impeller] Use MSAA attachments for offscreen passes

### PR DESCRIPTION
**Pull request by bdero**
Fixes https://github.com/flutter/flutter/issues/104759.

When loading an MSAA color attachment, it's not possible to pull the samples from the initial resolve texture, and so storing the MSAA texture is unfortunately necessary.
All of our shaders currently don't bind/read MSAA textures, so sometimes we also need to store the resolve texture right now.

Follow-up work:
- There are many cases where we can eliminate textures being created or disable the resolve. I need to come back around and add some smarts in `EntityPass` to make this happen.
- Make a builder for `RenderPass`.
- Offscreen MSAA textures are not possible in GLES, so offscreen MSAA passes need to be gracefully disabled.

### CODE REVIEW

File: impeller/entity/contents/content_context.h [hunk 1]
```diff
@@ -337,6 +337,7 @@ class ContentContext {
   /// @brief  Creates a new texture of size `texture_size` and calls
   ///         `subpass_callback` with a `RenderPass` for drawing to the texture.
   std::shared_ptr<Texture> MakeSubpass(ISize texture_size,
+                                       bool enable_msaa,
```
Comment (zanderso) [inline impeller/entity/contents/content_context.h:340]: Maybe use an enum so this can be self-documenting? Like `kMSAAEnabled`, `kMSAADisabled`.

**Review by zanderso — APPROVED** [review]
Reducing memory footprint in follow-up work sgtm.

lgtm w/ optional nit after fixing the build issue.

File: impeller/entity/contents/content_context.h (hunk 1 above)
Comment (bdero) [reply impeller/entity/contents/content_context.h]: Removed this param.

**Review by zanderso — APPROVED** [review]
still lgtm

**Pull request merged**